Mathlib/Algebra/Order/GroupWithZero/Lex.lean

```diff
@@ -0,0 +1,148 @@
+    [DecidablePred fun x : N₀ ↦ x = 0] : Monotone (inr M₀ N₀) := by
+  refine (WithZero.map'_mono MonoidHom.inr_mono).comp ?_
+  intro x y
+  obtain rfl | ⟨x, rfl⟩ := GroupWithZero.eq_zero_or_unit x <;>
+  obtain rfl | ⟨y, rfl⟩ := GroupWithZero.eq_zero_or_unit y <;>
+  · simp [WithZero.zero_le, WithZero.withZeroUnitsEquiv]
+
+lemma inr_strictMono [GroupWithZero M₀] [PartialOrder M₀] [LinearOrderedCommGroupWithZero N₀]
+    [DecidablePred fun x : N₀ ↦ x = 0] : StrictMono (inr M₀ N₀) :=
+  inr_mono.strictMono_of_injective inr_injective
+
+lemma fst_mono [LinearOrderedCommGroupWithZero M₀] [GroupWithZero N₀] [Preorder N₀] :
+    Monotone (fst M₀ N₀) := by
+  refine WithZero.forall.mpr ?_
+  simp +contextual [WithZero.forall, Prod.le_def]
+
+
+lemma snd_mono [GroupWithZero M₀] [Preorder M₀] [LinearOrderedCommGroupWithZero N₀] :
+    Monotone (snd M₀ N₀) := by
+  refine WithZero.forall.mpr ?_
+  simp [WithZero.forall, Prod.le_def]
+
+end MonoidWithZeroHom
+
+namespace LinearOrderedCommGroupWithZero
+
+variable (α β : Type*) [LinearOrderedCommGroupWithZero α] [LinearOrderedCommGroupWithZero β]
+
+open MonoidWithZeroHom
+
+/-- Given linearly ordered groups with zero M, N, the natural inclusion ordered homomorphism from
+M to `WithZero (Mˣ ×ₗ Nˣ)`, which is the linearly ordered group with zero that can be identified
+as their product. -/
+@[simps!]
+nonrec def inl : α →*₀o WithZero (αˣ ×ₗ βˣ) where
+  __ := (WithZero.map' (toLexMulEquiv ..).toMonoidHom).comp (inl α β)
+  monotone' := by simpa using (WithZero.map'_mono (Prod.Lex.toLex_mono)).comp inl_mono
+
+/-- Given linearly ordered groups with zero M, N, the natural inclusion ordered homomorphism from
+N to `WithZero (Mˣ ×ₗ Nˣ)`, which is the linearly ordered group with zero that can be identified
+as their product. -/
+@[simps!]
+nonrec def inr : β →*₀o WithZero (αˣ ×ₗ βˣ) where
+  __ := (WithZero.map' (toLexMulEquiv ..).toMonoidHom).comp (inr α β)
+  monotone' := by simpa using (WithZero.map'_mono (Prod.Lex.toLex_mono)).comp inr_mono
+
+/-- Given linearly ordered groups with zero M, N, the natural projection ordered homomorphism from
+`WithZero (Mˣ ×ₗ Nˣ)` to M, which is the linearly ordered group with zero that can be identified
+as their product. -/
+@[simps!]
+nonrec def fst : WithZero (αˣ ×ₗ βˣ) →*₀o α where
+  __ := (fst α β).comp (WithZero.map' (toLexMulEquiv ..).symm.toMonoidHom)
+  monotone' := by
+    -- this can't rely on `Monotone.comp` since `ofLex` is not monotone
+    intro x y
+    cases x <;>
+    cases y
+    · simp
+    · simp
+    · simp [WithZero.zero_lt_coe]
+    · simpa using Prod.Lex.monotone_fst _ _
+
+@[simp]
+theorem fst_comp_inl : (fst _ _).comp (inl α β) = .id α := by
+  ext x
+  obtain rfl | ⟨_, rfl⟩ := GroupWithZero.eq_zero_or_unit x <;>
+  simp
+
+theorem inlₗ_mul_inrₗ_eq_coe_toLex {m : α} {n : β} (hm : m ≠ 0) (hn : n ≠ 0) :
+    (inl α β m * inr α β n) = toLex (Units.mk0 _ hm, Units.mk0 _ hn) := by
```

Comment:
```suggestion
    inl α β m * inr α β n = toLex (Units.mk0 _ hm, Units.mk0 _ hn) := by
```
Is there no `inrₗ_mul_inlₗ` version?

Reply:
Not sure what that would give differently, since this is over a commgroup